### PR TITLE
fix: sfheader mobile navigation

### LINKED ISSUE #1635: [BUG] SfHeader mobile navigation broken view
**Describe the bug**
When on mobile view there is no items visible when navigation modifier is on.

**How to reproduce**
Go to the SfHeader storybook and choose css modifier - sf-header-has-mobile-navigation. 
There are only chevrons appearing without items.

**Expected behavior**
All menu items should appear when  sf-header-has-mobile-navigation is on.

**Actual behavior**
image

**Code examples**

**Screenshots**

**Devices (please complete the following information):**
 - Device: [e.g. iPhone6, Desktop]
 - Browser: [e.g chrome, safari][version]

**Additional context**
Anything you would like to add but was not sure which section it fits. For example: tech stack used (Nuxt, Vue, Gridsome, React, etc)

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue
Closes #1635

# Scope of work
Fix style for mobile navigation modifier and isNavVisable prop. 
Add menuItem for mobile to story. 

# Screenshots of visual changes
image

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/shared/styles/components/organisms/SfHeader.scss [hunk 1]
```diff
@@ -151,18 +152,21 @@
...
   }
   &__navigation {
-    order: 1;
-    display: var(--header-navigation-display, none);
+    order: 1;    
     flex: 0 0 calc(100% + var(--spacer-sm) * 2);
     margin: var(--header-navigation-margin, 0 calc(var(--spacer-sm) * -1));
     @include for-desktop {
-      --header-navigation-display: flex;
+      display: var(--header-navigation-display, flex);
```
Comment (mayashavin) [inline packages/shared/styles/components/organisms/SfHeader.scss:159]: remove `--header-navigation-display`, use `flex` only

File: packages/vue/src/components/organisms/SfHeader/SfHeader.stories.js [hunk 2]
```diff
@@ -144,6 +145,10 @@ const Template = (args, { argTypes }) => ({
       <template slot="desktop-navigation-item">
         <SfLink href="#">{{item}}</SfLink>
       </template>
+      <template slot="mobile-navigation-item">
```
Comment (mayashavin) [inline packages/vue/src/components/organisms/SfHeader/SfHeader.stories.js:148]: Why do we want `SfMenuItem`? Any specific reason? If nothing special, please use `SfLink` is enough

**Review by mayashavin — CHANGES_REQUESTED** [review]

**Comment by AdamPawlinski** [comment]
I changed base branch to develop as this change is not so relevant to be included in hotfix.

**Comment by mayashavin** [comment]
Will merge once v.10.1 is merged

**Review by mayashavin — DISMISSED** [review]

**Review by mayashavin — APPROVED** [review]

**Pull request merged**